From ef0d2a151646fb5e487b9bf000f49c9e60362800 Mon Sep 17 00:00:00 2001
From: Jian He <jianhe@apache.org>
Date: Fri, 14 Nov 2014 17:24:09 -0800
Subject: [PATCH 724/791] YARN-2432. RMStateStore should process the pending
 events before close. Contributed by Varun Saxena
 (cherry picked from commit
 9b860661a5754e829bd8c34ef092757421589bbe)

(cherry picked from commit 21054c5d149ae56ee998e42a11cafd80b7aaa279)
---
 .../resourcemanager/recovery/RMStateStore.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/RMStateStore.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/RMStateStore.java
index 68a084d..b801ab4 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/RMStateStore.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/RMStateStore.java
@@ -488,8 +488,8 @@ protected void serviceStart() throws Exception {
 
   @Override
   protected void serviceStop() throws Exception {
-    closeInternal();
     dispatcher.stop();
+    closeInternal();
   }
 
   /**
-- 
1.7.9.5

